### ISSUE TITLE
pt to engine error

### ISSUE THREAD

**Issue by omaiyiwa** [ultralytics/yolov5#10851]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Export

### Bug

export: data=D:\yolo\yolov5-master\data\own_data.yaml, weights=D:\yolo\yolov5-master\runs\train\exp97\weights\best.pt, imgsz=[640, 640], batch_size=1, device=0, half=False, inplace=False, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=12, verbose=False, workspace=4, nms=False, agnostic_nms=False, topk_per_class=100, topk_all=100, iou_thres=0.45, conf_thres=0.25, include=['engine']
YOLOv5  2022-9-16 Python-3.8.0 torch-1.12.1+cu116 CUDA:0 (NVIDIA GeForce RTX 3090 Ti, 24563MiB)

Fusing layers... 
YOLOv5x summary: 444 layers, 86543509 parameters, 0 gradients, 205.2 GFLOPs

PyTorch: starting from D:\yolo\yolov5-master\runs\train\exp97\weights\best.pt with output shape (1, 25200, 61) (165.8 MB)

False starting export with onnx 1.12.0...
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
ONNX: export success  24.4s, saved as D:\yolo\yolov5-master\runs\train\exp97\weights\best.onnx (330.6 MB)

TensorRT: starting export with TensorRT 8.2.5.1...
[01/29/2023-09:08:17] [TRT] [I] [MemUsageChange] Init CUDA: CPU +454, GPU +0, now: CPU 12392, GPU 3866 (MiB)
[01/29/2023-09:08:18] [TRT] [I] [MemUsageSnapshot] Begin constructing builder kernel library: CPU 12456 MiB, GPU 3866 MiB
[01/29/2023-09:08:18] [TRT] [I] [MemUsageSnapshot] End constructing builder kernel library: CPU 12622 MiB, GPU 3910 MiB
[01/29/2023-09:08:18] [TRT] [I] ----------------------------------------------------------------
[01/29/2023-09:08:18] [TRT] [I] Input filename:   D:\yolo\yolov5-master\runs\train\exp97\weights\best.onnx
[01/29/2023-09:08:18] [TRT] [I] ONNX IR version:  0.0.7
[01/29/2023-09:08:18] [TRT] [I] Opset version:    13
[01/29/2023-09:08:18] [TRT] [I] Producer name:    pytorch
[01/29/2023-09:08:18] [TRT] [I] Producer version: 1.12.1
[01/29/2023-09:08:18] [TRT] [I] Domain:           
[01/29/2023-09:08:18] [TRT] [I] Model version:    0
[01/29/2023-09:08:18] [TRT] [I] Doc string:       
[01/29/2023-09:08:18] [TRT] [I] ----------------------------------------------------------------
[01/29/2023-09:08:19] [TRT] [W] onnx2trt_utils.cpp:366: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
[01/29/2023-09:08:19] [TRT] [E] ModelImporter.cpp:773: While parsing node number 313 [Resize -> "onnx::Concat_568"]:
[01/29/2023-09:08:19] [TRT] [E] ModelImporter.cpp:774: --- Begin node ---
[01/29/2023-09:08:19] [TRT] [E] ModelImporter.cpp:775: input: "input.684"
input: ""
input: "onnx::Resize_1183"
output: "onnx::Concat_568"
name: "Resize_313"
op_type: "Resize"
attribute {
  name: "coordinate_transformation_mode"
  s: "asymmetric"
  type: STRING
}
attribute {
  name: "cubic_coeff_a"
  f: -0.75
  type: FLOAT
}
attribute {
  name: "mode"
  s: "nearest"
  type: STRING
}
attribute {
  name: "nearest_mode"
  s: "floor"
  type: STRING
}

[01/29/2023-09:08:19] [TRT] [E] ModelImporter.cpp:776: --- End node ---
[01/29/2023-09:08:19] [TRT] [E] ModelImporter.cpp:779: ERROR: builtin_op_importers.cpp:3609 In function importResize:
[8] Assertion failed: scales.is_weights() && "Resize scales must be an initializer!"
failed to load ONNX file: D:\yolo\yolov5-master\runs\train\exp97\weights\best.onnx
TensorRT: export failure  26.2s: cannot unpack non-iterable NoneType object

### Environment

Win10  YOLOv5  2022-9-16 Python-3.8.0 torch-1.12.1+cu116 CUDA:0 (NVIDIA GeForce RTX 3090 Ti, 24563MiB)

### Minimal Reproducible Example

I downloaded the latest project file from github without making any changes. When I transferred my pt file to engine, false starting export with onnx 1.12.0... appeared, but the onnx file was successfully exported，In the next process of turning to engine, in line 258 of export.py, assert onnx. exists(), f'failed to export ONNX file: {onnx} 'made an error, but the onnx file was successfully exported. In order to clearly understand what happened, I annotate the 258 lines of statements and use try... except...， The result is not unpack non-interable NoneType object.

### Additional

The model file can be successfully exported three months ago. At that time, it was 2060 graphics card. Now it is 3090ti. detect.py can successfully run the pt and onnx file.

detect: weights=D:\yolo\yolov5-master\runs\train\exp97\weights\best.onnx, source=D:\yolo\yolov5-master\data\images, data=D:\yolo\yolov5-master\data\own_data.yaml, imgsz=[640, 640], conf_thres=0.5, iou_thres=0.45, max_det=1000, device=, view_img=False, save_txt=False, save_conf=False, save_crop=False, nosave=False, classes=None, agnostic_nms=False, augment=False, visualize=False, update=False, project=runs\detect, name=exp, exist_ok=False, line_thickness=3, hide_labels=False, hide_conf=False, half=False, dnn=False, vid_stride=1
YOLOv5  2022-9-16 Python-3.8.0 torch-1.12.1+cu116 CUDA:0 (NVIDIA GeForce RTX 3090 Ti, 24563MiB)

Loading D:\yolo\yolov5-master\runs\train\exp97\weights\best.onnx for ONNX Runtime inference...
image 1/1 D:\yolo\yolov5-master\data\images\8.png: 640x640 *** 35.9ms
Speed: 949.6ms pre-process, 35.9ms inference, 43.4ms NMS per image at shape (1, 3, 640, 640)
Results saved to runs\detect\exp

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by omaiyiwa** [comment]
Sorry, my version of TensorRT does not correspond to the version of pystore

**Comment by glenn-jocher** [comment]
@omaiyiwa sorry to hear you're encountering issues with TensorRT. There could be compatibility issues between the specific versions of TensorRT, PyTorch, and the YOLOv5 repository. Have you considered updating your environment to ensure compatibility among the software components? You can find more information on system requirements and compatibility in the Ultralytics YOLOv5 documentation at https://docs.ultralytics.com/yolov5/. Let me know if you need further assistance with this.
